File: backend/src/main/java/com/castify/backend/models/comment/CommentModel.java
package com.castify.backend.models.comment;

import com.castify.backend.models.user.UserSimple;
import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@AllArgsConstructor
@NoArgsConstructor
@Data
public class CommentModel {
    private String id;
    private String parentId;
    private String content;
    private String mentionedUser;
    private long totalLikes;
    private long totalReplies;
    @JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm:ss.SSS")
    private LocalDateTime timestamp;

    private boolean isLiked;
    private UserSimple user;
}

File: views/src/models/CommentModel.ts
import { UserSimple } from "./User";

export interface Comment {
  id: string;
  parentId: string;
  content: string;
  mentionedUser: string;
  totalLikes: number;
  totalReplies: number;
  timestamp: string;
  lastModified: string;
  user: UserSimple;
  liked: boolean;
  replies?: Comment[];
}